src/components/LocationSelector/LocationSelector.tsx

```diff
@@ -0,0 +1,140 @@
+import { SyntheticEvent, useState, useMemo, useEffect } from 'react';
+import Autocomplete, {
+  AutocompleteChangeReason,
+} from '@mui/material/Autocomplete';
+import TextField from '@mui/material/TextField';
+import CircularProgress from '@mui/material/CircularProgress';
+import debounce from 'lodash/debounce';
+import pick from 'lodash/pick';
+import { useSnackbar } from 'notistack';
+import { axiosDefaultConfig } from 'api/axiosDefaultConfig';
+import Location from 'components/LocationSelector/location';
+import { COUNTRY_CODES } from './countryCodes';
+import * as classes from './styles';
+
+const DEBOUNCE_DELAY = 400;
+const MIN_INPUT_LEN = 2;
+const LIMIT_PARAM_VALUE = 5;
+const ERROR_MESSAGE =
+  'Sorry, we could not find any results that match your search.';
+const SPINNER_SIZE = 25;
+
+interface Props {
+  selectedLocation: Location | null;
+  // eslint-disable-next-line no-unused-vars
+  onSelectedLocation: (value: Location) => void;
+}
+
+export default function LocationSelector({
+  selectedLocation,
+  onSelectedLocation,
+}: Props) {
+  const { enqueueSnackbar } = useSnackbar();
+  const [inputValue, onInputValue] = useState<string>('');
+  const [suggestions, onSuggestions] = useState<Location[]>([]);
+  const [isOpen, onIsOpen] = useState(false);
+  const [isLoading, onIsLoading] = useState(false);
+
+  const fetchGeoData = useMemo(
+    () =>
+      debounce(async () => {
+        if (inputValue === '' || inputValue.length <= MIN_INPUT_LEN) {
+          onSuggestions([]);
+          onIsLoading(false);
+        } else {
+          try {
+            onIsLoading(true);
+            const response = await axiosDefaultConfig.get<Location[]>(
+              '/geo/1.0/direct',
+              {
+                params: {
+                  q: inputValue,
+                  limit: LIMIT_PARAM_VALUE,
+                },
+              },
+            );
+            const locationData = response.data.map((location) =>
+              pick(location, ['name', 'state', 'country', 'lat', 'lon']),
+            );
+
+            onSuggestions(locationData);
+          } catch (error) {
+            console.log(error);
+            enqueueSnackbar(ERROR_MESSAGE, { variant: 'error' });
+          } finally {
+            onIsLoading(false);
+          }
+        }
+      }, DEBOUNCE_DELAY),
+    [enqueueSnackbar, inputValue],
```

Comment:
Your debounce won't work. Accept input value as an argument and call this function with `inputValue` in effect